File: examples/python/basic.py
"""basic.py — i18nme Python examples

Usage:
    export I18N_API_KEY="key_live_XXXX"
    python basic.py

Requires: pip install httpx
"""

import os
import httpx

BASE_URL = "https://sapi.i18nme.com"
API_KEY  = os.environ.get("I18N_API_KEY", "")

if not API_KEY:
    raise EnvironmentError("Set the I18N_API_KEY environment variable first.")

headers = {"X-API-Key": API_KEY}

client = httpx.Client(headers=headers, timeout=10)

# ── Health check (no auth required) ──────────────────────────────────────────
health = httpx.get(f"{BASE_URL}/health")
print("health:", health.json())
# → {'status': 'ok'}

# ── Manifest — available languages ───────────────────────────────────────────
manifest = client.get(f"{BASE_URL}/v1/cached/manifest").json()
languages = manifest["languages"]
print("languages:", [l["code"] for l in languages])
# → ['en', 'de', 'pl']

# ── All translations for a language ──────────────────────────────────────────
translations = client.get(f"{BASE_URL}/v1/cached/translations/en").json()
print("translations:", translations)
# → {'common': {'welcome': 'Welcome'}, 'nav': {'home': 'Home'}}

# ── Translations for a single namespace ──────────────────────────────────────
common = client.get(f"{BASE_URL}/v1/cached/translations/en/common").json()
print("common:", common)
# → {'welcome': 'Welcome', 'logout': 'Log out'}

# ── Single translation key ────────────────────────────────────────────────────
key = client.get(f"{BASE_URL}/v1/cached/translations/en/common/welcome").json()
print("key value:", key["value"])
# → 'Welcome'

# ── Fetch all languages in parallel (httpx async) ────────────────────────────
import asyncio

async def fetch_all():
    async with httpx.AsyncClient(headers=headers, timeout=10) as aclient:
        tasks = [
            aclient.get(f"{BASE_URL}/v1/cached/translations/{l['code']}")
            for l in languages
        ]
        responses = await asyncio.gather(*tasks)
    return {
        lang["code"]: res.json()
        for lang, res in zip(languages, responses)
    }

all_bundles = asyncio.run(fetch_all())
print("fetched locales:", list(all_bundles.keys()))

File: examples/javascript/basic.js
// basic.js — i18nme JavaScript examples
// Requires Node 18+ (built-in fetch). No extra packages needed.
//
// Usage:
//   export I18N_API_KEY="key_live_XXXX"
//   node basic.js

const BASE_URL = 'https://sapi.i18nme.com'
const API_KEY  = process.env.I18N_API_KEY

if (!API_KEY) {
  console.error('Set the I18N_API_KEY environment variable first.')
  process.exit(1)
}

const headers = { 'X-API-Key': API_KEY }

// ── Health check (no auth required) ──────────────────────────────────────────

const health = await fetch(`${BASE_URL}/health`)
console.log('health:', await health.json())
// → { status: 'ok' }

// ── Manifest — available languages ───────────────────────────────────────────

const manifestRes = await fetch(`${BASE_URL}/v1/cached/manifest`, { headers })
const { languages } = await manifestRes.json()
console.log('languages:', languages.map(l => l.code))
// → ['en', 'de', 'pl']

// ── All translations for a language ──────────────────────────────────────────

const transRes = await fetch(`${BASE_URL}/v1/cached/translations/en`, { headers })
const translations = await transRes.json()
console.log('translations:', translations)
// → { common: { welcome: 'Welcome', logout: 'Log out' }, nav: { home: 'Home' } }

// ── Translations for a single namespace ──────────────────────────────────────

const groupRes = await fetch(`${BASE_URL}/v1/cached/translations/en/common`, { headers })
const group = await groupRes.json()
console.log('common group:', group)
// → { welcome: 'Welcome', logout: 'Log out' }

// ── Single translation key ────────────────────────────────────────────────────

const keyRes = await fetch(`${BASE_URL}/v1/cached/translations/en/common/welcome`, { headers })
const key = await keyRes.json()
console.log('single key:', key.value)
// → 'Welcome'

// ── Fetch all languages in parallel ──────────────────────────────────────────

const allBundles = Object.fromEntries(
  await Promise.all(
    languages.map(async ({ code }) => {
      const res = await fetch(`${BASE_URL}/v1/cached/translations/${code}`, { headers })
      return [code, await res.json()]
    })
  )
)
console.log('all bundles:', Object.keys(allBundles))
// → ['en', 'de', 'pl']
